src/Tools/Source/RunTests/Cache/CachingTestExecutor.cs

```diff
@@ -55,15 +55,19 @@ internal CachingTestExecutor(Options options, ITestExecutor testExecutor, IDataS
         /// <returns></returns>
         private static TestResult Migrate(TestResult testResult)
         {
-            if (string.IsNullOrEmpty(testResult.ResultsFilePath))
```

Comment:
Is null possible or was this just checking for empty string?  If null is possible, is a null check required below?